ext/proxy/Dxxxx.md

```diff
@@ -1135,63 +1089,91 @@ Change 24.6 "Header `<experimental/ranges/iterator>` synopsis" ([iterator.synops
-> >   requires !Swappable<reference_t<_R1>, reference_t<_R2>>()
-> >     && IndirectlyMovable<_R1, _R2>() && IndirectlyMovable<_R2, _R1>()
-> > void iter_swap(I1&& r1, I2&& r2)
-> >   noexcept(is_nothrow_indirectly_movable<_R1, _R2>::value &&
-> >            is_nothrow_indirectly_movable<_R2, _R1>::value);
-> > ```
+> 24.6.? *iter_swap*                     [iterator.utils.iter_swap]
 >
-> 5\. *Effects*: Exchanges values referred to by two `Readable` objects.
+> 1\. The name `iter_swap` denotes a *customization point object* (17.5.2.1.5 [customization.point.object]). The effect of the expression `ranges::iter_swap(E1, E2)` for some expressions `E1` and `E2` is equivalent to `(void)iter_swap(E1, E2)`, with overload resolution performed in an unspecified context described below:
 >
-> 6\. \[*Example:* Below is a possible implementation:
-> > ```c++
-> > value_type_t<_R1> tmp(iter_move(r1));
-> > *r1 = iter_move(r2);
-> > *r2 = std::move(tmp);
-> > ```
+> 2\. The context in which the expression `(void)iter_swap(E1, E2)` is evaluted
 >
-> -- *end example*\]
-
+> > -- does not include a declaration of `ranges::iter_swap`,
+> >
+> > -- includes the lookup set produced by argument-dependent lookup ([basic.lookup.argdep]), and
+> >
+> > -- includes `iter_swap` function template(s) equivalent to the following:
+> >
+> > > ```c++
+> > > // Exposition only
+> > > template <class I1, class I2>
+> > > void iter_swap(I1&&, I2&&) = delete;
+> > >
+> > > // Exposition only
+> > > template <class I1, class I2,
+> > >   Readable _R1 = remove_reference_t<I1>,
+> > >   Readable _R2 = remove_reference_t<I2>>
+> > >   requires Swappable<reference_t<_R1>, reference_t<_R2>>()
+> > > void iter_swap(I1&& r1, I2&& r2)
+> > >   noexcept(is_nothrow_swappable_with_v<reference_t<_R1>, reference_t<_R2>>);
+> > > ```
+> > > > 2.1. *Effects:* Equivalent to `ranges::swap(*r1, *r2)`
+> > >
+> > > ```c++
+> > > // Exposition only
+> > > template <class I1, class I2,
+> > >   Readable _R1 = std::remove_reference_t<I1>,
+> > >   Readable _R2 = std::remove_reference_t<I2>>
+> > >   requires !Swappable<reference_t<_R1>, reference_t<_R2>>()
+> > >     && IndirectlyMovable<_R1, _R2>() && IndirectlyMovable<_R2, _R1>()
+> > > void iter_swap(I1&& r1, I2&& r2)
+> > >   noexcept(is_nothrow_indirectly_movable_v<_R1, _R2> &&
+> > >            is_nothrow_indirectly_movable_v<_R2, _R1>);
+> > > ```
+> > > > 2.2. *Effects*: Exchanges values referred to by two `Readable` objects.
+> > > >
+> > > > 2.3. \[*Example:* Below is a possible implementation:
+> > > > > ```c++
+> > > > > value_type_t<_R1> tmp(iter_move(r1));
+> > > > > *r1 = iter_move(r2);
+> > > > > *r2 = std::move(tmp);
+> > > > > ```
+> > > > >
+> > > > -- *end example*\]
+>
+> 3\. If the `iter_swap` function selected by overload resolution ([over.match]) does not swap the values of the objects to which the expressions `E1` and `E2` refer, the program is ill-formed with no diagnostic required.
```

Comment:
proxy references don't necessarily refer to objects; I'd rephrase this. Maybe "swap the values denoted by the expressions `E1` and `E2`.